moped-editor/src/views/dashboard/UserSavedViewsTable.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useMemo, useEffect, useCallback } from "react";
+import { useQuery, useMutation } from "@apollo/client";
+
+// Material
+import { CircularProgress } from "@mui/material";
+import { DataGridPro, GridRowModes } from "@mui/x-data-grid-pro";
+import Link from "@mui/material/Link";
+
+import ApolloErrorHandler from "src/components/ApolloErrorHandler";
+import {
+  USER_SAVED_VIEWS_QUERY,
+  UPDATE_USER_SAVED_VIEW,
+  DELETE_USER_SAVED_VIEW,
+} from "src/queries/userSavedViews";
+import { getSessionDatabaseData } from "src/auth/user";
+
+import dataGridProStyleOverrides from "src/styles/dataGridProStylesOverrides";
+import DataGridActions from "src/components/DataGridPro/DataGridActions";
+import DataGridTextField from "src/components/DataGridPro/DataGridTextField";
+import DeleteConfirmationModal from "../projects/projectView/DeleteConfirmationModal";
+import { FormattedDateString } from "src/utils/dateAndTime";
+import config from "../../config";
+
+const requiredFields = ["description"];
+
+/** Hook that provides memoized column settings */
+const useColumns = ({
+  rowModesModel,
+  handleDeleteOpen,
+  handleSaveClick,
+  handleCancelClick,
+  handleEditClick,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "Description",
+        field: "description",
+        editable: true,
+        width: 500,
+        renderEditCell: (props) => <DataGridTextField {...props} />,
+      },
+      {
+        headerName: "URL",
+        field: "url",
+        editable: false,
+        width: 300,
+        renderCell: ({ row }) =>
+          row.url ? (
+            <Link
+              href={row.url}
+              sx={{
+                overflow: "hidden",
+                textOverflow: "ellipsis",
+                display: "block",
+              }}
+            >
+              {`${config.env.APP_CLOUDFRONT}${row.url}`}
```

Comment:
the `config.env.APP_CLOUDFRONT` threw me off at first when testing locally, i thought it was going to try to redirect me to a broken link, but I see that the APP_CLOUDFRONT is used only as what is displayed, and not part of the actual url